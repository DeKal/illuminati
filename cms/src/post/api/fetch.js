import { get, post } from 'core/api'

const fetchPostFromSlug = async slug => {
  return await get(`post/${slug}`)
}

const savingPost = async postData => {
  return await post('post/save', postData)
}

const createPost = async postData => {
  return await post('post/create', postData)
}

export { fetchPostFromSlug, savingPost, createPost }
