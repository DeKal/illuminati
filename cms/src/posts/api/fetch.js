import { get } from 'core/api'

const fetchPostSummary = async () => {
  return await get('post/all-post/summary')
}

export { fetchPostSummary }
