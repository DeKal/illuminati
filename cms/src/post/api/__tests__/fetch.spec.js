import { fetchPostFromSlug } from '../fetch'
import mockAxios from 'axios'
jest.mock('axios')

const post = {
  data: {
    id: '5e24614a68ac5fa5373bf91c',
    slug: 'design-patterns-tutorial',
    title: 'Design Patterns â€“ Example Tutorial',
    brief:
      'Design Patterns are very popular among software developers. A design pattern is a well-described solution to a common software problem. With this, we are cool.',
    time: '15/01/2020',
    tag: ['System Design'],
    author: 'Li Li',
    content: '<p><strong>Add new content here.</strong></p>',
    viewNum: 0,
    commentNum: 0,
    likeNum: 0
  },
  success: true
}

describe('fetchPostFromSlug ', () => {
  it('fetchPostFromSlug with mock request', async () => {
    //@ts-ignore
    mockAxios.get.mockResolvedValue({
      data: post
    })

    const data = await fetchPostFromSlug('slug')
    expect(data).toMatchSnapshot()
  })

  it('fetchPostFromSlug with error request', async () => {
    //@ts-ignore
    mockAxios.get.mockResolvedValue({
      data: {
        error: {
          code: 204,
          message: 'Cannot find post'
        },
        success: false
      }
    })

    try {
      await fetchPostFromSlug('slug')
    } catch (e) {
      expect(e).toEqual({
        code: 204,
        message: 'Cannot find post'
      })
    }
  })
})
