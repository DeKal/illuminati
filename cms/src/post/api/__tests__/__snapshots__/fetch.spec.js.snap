// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchPostFromSlug  fetchPostFromSlug with mock request 1`] = `
Object {
  "author": "Li Li",
  "brief": "Design Patterns are very popular among software developers. A design pattern is a well-described solution to a common software problem. With this, we are cool.",
  "commentNum": 0,
  "content": "<p><strong>Add new content here.</strong></p>",
  "id": "5e24614a68ac5fa5373bf91c",
  "likeNum": 0,
  "slug": "design-patterns-tutorial",
  "tag": Array [
    "System Design",
  ],
  "time": "15/01/2020",
  "title": "Design Patterns â€“ Example Tutorial",
  "viewNum": 0,
}
`;
